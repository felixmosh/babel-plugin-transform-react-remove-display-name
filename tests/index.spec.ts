import { transformFileSync } from '@babel/core';
import path from 'path';
import displayNameTransform from '../src/index';

const fixturesDir = path.join(__dirname, 'fixtures');

function fixtureAssert(testFilename: string, options = {}) {
  it(`should pass ${testFilename.split('-').join(' ').toLowerCase()}`, () => {
    const actualPath = path.join(fixturesDir, `${testFilename}.tsx`);

    const actual = transformFileSync(actualPath, {
      babelrc: false,
      plugins: [[displayNameTransform, options]],
      presets: ['@babel/preset-react'],
    })?.code;

    expect(actual).toMatchSnapshot();
  });
}

describe('remove displayName', () => {
  // fixtureAssert('function-component');
  // fixtureAssert('functional-component');
  // fixtureAssert('exotic-component');
  //
  // fixtureAssert('not-react-function-component');
  // fixtureAssert('object-literal');
  // fixtureAssert('other-property-name');
  // fixtureAssert('other-property-type');
  fixtureAssert('with-usage-of-hoc');
});
