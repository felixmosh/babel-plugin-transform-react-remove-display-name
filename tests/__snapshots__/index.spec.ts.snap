// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove displayName should pass exotic component 1`] = `
"export const ForwardRefComponent = React.forwardRef(() => {
  return /*#__PURE__*/React.createElement("div", null);
});
if (process.env.NODE_ENV !== "production") {
  ForwardRefComponent.displayName = 'ForwardRefComponent';
}
export const MemoComponent = React.memo(() => {
  return /*#__PURE__*/React.createElement("div", null);
});
if (process.env.NODE_ENV !== "production") {
  MemoComponent.displayName = 'MemoComponent';
}"
`;

exports[`remove displayName should pass function component 1`] = `
"export function FunctionComponent() {
  return /*#__PURE__*/React.createElement("div", null);
}
if (process.env.NODE_ENV !== "production") {
  FunctionComponent.displayName = 'FunctionComponent';
}"
`;

exports[`remove displayName should pass functional component 1`] = `
"export const FunctionalComponent = () => {
  return /*#__PURE__*/React.createElement("div", null);
};
if (process.env.NODE_ENV !== "production") {
  FunctionalComponent.displayName = 'FunctionalComponent';
}"
`;

exports[`remove displayName should pass not react function component 1`] = `
"export function notFunctionalComponent() {
  return 2;
}
notFunctionalComponent.displayName = 'FunctionalComponent';"
`;

exports[`remove displayName should pass object literal 1`] = `
"const data = {};
data.displayName = 'FunctionalComponent';"
`;

exports[`remove displayName should pass other property name 1`] = `
"export const FunctionalComponent = () => {
  return /*#__PURE__*/React.createElement("div", null);
};
FunctionalComponent.propTypes = 'FunctionalComponent';
function FunctionComponent() {}
FunctionComponent.propTypes = 'bla';"
`;

exports[`remove displayName should pass other property type 1`] = `
"export const FunctionalComponent = () => {
  return /*#__PURE__*/React.createElement("div", null);
};
FunctionalComponent.displayName = null;
function FunctionComponent() {
  return /*#__PURE__*/React.createElement("div", null);
}
FunctionComponent.displayName = {};"
`;
